javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Validator.java

```diff
@@ -0,0 +1,159 @@
+            .isSuccessful(result.isSuccessful())
+            .errorMessages(result.getMessages().stream()
+                .map(singleValidationMessage -> ValidatorErrorMessage.builder()
+                    .severity(singleValidationMessage.getSeverity().getCode())
+                    .msg(singleValidationMessage.getMessage())
+                    .build())
+                .collect(Collectors.toList())
+            )
+            .build();
+    }
+
+    private PrePopulatedValidationSupport loadIgs(final FhirContext ctx) {
+
+        final ImmutableSet<String> allowedResourceTypes = ImmutableSet.of("StructureDefinition", "CodeSystem", "ValueSet");
+
+        IParser parser = ctx.newJsonParser();
+        parser.setParserErrorHandler(new StrictErrorHandler());
+        final PrePopulatedValidationSupport prePopulatedValidationSupport = new PrePopulatedValidationSupport(ctx);
+
+        try (ScanResult allFiles = new ClassGraph().acceptPaths(IMPLEMENTATION_GUIDES_FOLDER).rejectPaths(IMPLEMENTATION_GUIDES_FOLDER + "/*/*").scan()) {
+            ResourceList jsonResources = allFiles.getResourcesWithExtension("json");
+
+            ResourceList indexFiles = jsonResources.filter(x -> x.getPath().endsWith(".index.json"));
+
+            for (Resource indexFile : indexFiles) {
+                IgIndex igIndex = GSON.fromJson(indexFile.getContentAsString(), IgIndex.class);
+                for (IgFile file : igIndex.files) {
+                    if (allowedResourceTypes.contains(file.resourceType)) {
+
+                        String igResourcePath = indexFile.getPath().replace(".index.json", file.filename);
+                        log.info("loading {}", igResourcePath);
+                        Resource resource = allFiles.getResourcesWithPath(igResourcePath).get(0);
+                        try (InputStream inputStream = resource.open()) {
+                            switch (file.resourceType) {
+                                case "StructureDefinition":
+                                    prePopulatedValidationSupport.addStructureDefinition(parser.parseResource(StructureDefinition.class, inputStream));
+                                    break;
+                                case "CodeSystem":
+                                    prePopulatedValidationSupport.addCodeSystem(parser.parseResource(CodeSystem.class, inputStream));
+                                    break;
+                                case "ValueSet":
+                                    prePopulatedValidationSupport.addValueSet(parser.parseResource(ValueSet.class, inputStream));
+                                    break;
+                                default:
+                                    // cannot happen since we checked for allowedResourceTypes
+                                    break;
+                            }
+                        }
+                    }
+                }
+            }
+        } catch (IOException e) {
```

Comment:
If we can't read one of the files, should we error out here, and stop execution or should we only report an error happened?

What happens if the validator expects to load in a file, but fails to do it. Will the validator not validate against that IG?

Reply:
I changed it to just rethrow the exception. 

Since we are using provisioned concurrency, failing to load an IG will be an uncaught exception that will make the deployment fail. This is an ok behavior since that means that the IG files are wrong and you need to fix them and redeploy.